packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -0,0 +1,150 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:process/process.dart';
+
+import '../../artifacts.dart';
+import '../../base/error_handling_io.dart';
+import '../../base/file_system.dart';
+import '../../base/io.dart';
+import '../../base/logger.dart';
+import '../../flutter_manifest.dart';
+import '../build_system.dart';
+
+/// Applies a series of user-specified asset-transforming packages to an asset file.
+final class AssetTransformer {
+  AssetTransformer({
+    required ProcessManager processManager,
+    required Logger logger,
+    required FileSystem fileSystem,
+    required Artifacts artifacts,
+  })  : _processManager = processManager,
+        _logger = logger,
+        _fileSystem = fileSystem {
+          _dartBinary = artifacts.getArtifactPath(Artifact.engineDartBinary);
+        }
+
+  final ProcessManager _processManager;
+  final Logger _logger;
+  final FileSystem _fileSystem;
+  late String _dartBinary;
+  final Random _random = Random();
+
+  /// The [Source] inputs that targets using this should depend on.
+  ///
+  /// See [Target.inputs].
+  static const List<Source> inputs = <Source>[
+    Source.pattern(
+      '{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/asset_transformer.dart',
+    ),
+  ];
+
+  /// Applies, in sequence, a list of transformers to an [asset] and then copies
+  /// the output to [outputPath].
+  ///
+  /// Returns `true` if successful and `false` otherwise.
+  Future<bool> transformAsset({
+    required File asset,
+    required String outputPath,
+    required String workingDirectory,
+    required List<AssetTransformerEntry> transformerEntries,
+  }) async {
+
+    String getTempFilePath() => '${_random.nextInt(1<<32)}${_fileSystem.path.extension(asset.path)}';
```

Comment:
for context my concern about it being random would be it makes debugging issues here difficult.

Reply:
I guess because we re-assign tempInputFile and tempOutputFile. What if we instead appended an integer to the original asset name, representing the iteration number we're in?

Reply:
In the case where an asset has multiple chained transformers, we need unique output filenames to be able to tell whether or not a transformer produced an output.
If it's of any value, we could use sequenced names (i.e. transform1, transform2, etc.), instead.

Reply:
yeah, i would prefer something deterministic, and even better a scheme that is easily human understandable :)

Reply:
also, 32-bit pseudo random ints will eventually get collisions :)